Adds module code (#1)
hashicorp/terraform-aws-hashicorp-vault-eks-addon: README.md

# HashiCorp Vault Add-on for AWS EKS

> Deploy HashiCorp Vault to AWS EKS Clusters with this add-on.

## Table of Contents

- [HashiCorp Vault Addon for AWS EKS](#hashicorp-vault-addon-for-aws-eks)
  - [Table of Contents](#table-of-contents)
  - [Usage](#usage)
  - [Author Information](#author-information)
  - [License](#license)

## Usage

If you would like to override any defaults with the chart, you can pass it via the `helm_config` variable.

### Unsealing Vault

> Once the add-on has been deployed, the Vault server can be unsealed using the following commands.

Reviewer: ```suggestion
Once the add-on has been deployed, you can unseal the Vault server using the following commands.
```

> You will need to be in the `vault` (Kubernetes) namespace while running these commands, by default

> You will first need to initialize the Vault server:

Reviewer: ```suggestion
First, initialize the Vault server. This will return [unseal keys](https://www.vaultproject.io/docs/concepts/seal#seal-unseal) and [root token](https://www.vaultproject.io/docs/concepts/tokens#root-tokens). Store this in a safe place since you will use them to unseal the Vault server.
```

```sh
kubectl exec -it vault-0 -n vault -- vault operator init
```

> 
> Take note of the [unseal keys](https://www.vaultproject.io/docs/concepts/seal#seal-unseal) and [root token](https://www.vaultproject.io/docs/concepts/tokens#root-tokens) that get generated.

Reviewer: ```suggestion
```

Next, unseal the Vault server by providing at least _3_ of these keys to unseal Vault before servicing requests.

```sh
kubectl exec -it vault-0 -n vault -- vault operator unseal <key 1>
kubectl exec -it vault-0 -n vault -- vault operator unseal <key 2>
kubectl exec -it vault-0 -n vault -- vault operator unseal <key 3>
 ```

> Confirm that the Vault server is unsealed by checking the status of the Vault server:
> 
> ```sh
> kubectl get pods -n vault | grep vault
> 

Reviewer: ````suggestion
Confirm that the Vault server is unsealed by checking the status of the Vault server.

```sh
kubectl get pods -n vault | grep vault
```

This command will return a similar output to the following.

```sh

````

Reviewer: this way, users can c/p the `kubectl` command by clicking the button

Author: Love it!

NAME                 | READY | STATUS  | RESTARTS | AGE
---------------------|-------|---------|----------|-----
vault-0              |  1/1  | Running | 0        | 28m
vault-agent-injector |  1/1  | Running | 0        | 1m
```

> At this point, Vault can be used to store, access and deploy secrets to your application workloads.

Reviewer: active voice

```suggestion
At this point, you can use Vault to store, access and deploy secrets to your application workloads.
```

> See [this guide](https://learn.hashicorp.com/tutorials/vault/getting-started-first-secret?in=vault/getting-started) for a detailed overview on how to get started.

Reviewer: able-ist lang

```suggestion
Refer to [this guide](https://learn.hashicorp.com/tutorials/vault/getting-started-first-secret?in=vault/getting-started) for a detailed overview on how to get started.
```
